softlayer/resource_softlayer_network_vlan.go

```diff
@@ -0,0 +1,419 @@
+package softlayer
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/helpers/hardware"
+	"github.com/softlayer/softlayer-go/helpers/location"
+	"github.com/softlayer/softlayer-go/helpers/product"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+	"github.com/softlayer/softlayer-go/sl"
+	"log"
+	"time"
+)
+
+const (
+	AdditionalServicesPackageType            = "ADDITIONAL_SERVICES"
+	AdditionalServicesNetworkVlanPackageType = "ADDITIONAL_SERVICES_NETWORK_VLAN"
+)
+
+func resourceSoftLayerNetworkVlan() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceSoftLayerNetworkVlanCreate,
+		Read:     resourceSoftLayerNetworkVlanRead,
+		Update:   resourceSoftLayerNetworkVlanUpdate,
+		Delete:   resourceSoftLayerNetworkVlanDelete,
+		Exists:   resourceSoftLayerNetworkVlanExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"datacenter": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"primary_subnet_size": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"primary_router_hostname": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
```

Comment:
I see.  That makes sense.  And because it's `ForceNew: true`, we never have to deal with the case where the config is updated.  Nice.